File: client/src/types/sys/Menu.ts
export type MenuItem = {
  id: number
  viName: string
  enName: string
  path: string
  icon?: string
  displayOrder: number
  isActive: boolean
  parentId?: number
  sysmodule?: string
  isGroup: boolean
  children?: MenuItem[]
  [key: string]: any
}

export const defaultMenuItem: MenuItem = {
  id: 0,
  viName: '',
  enName: '',
  path: '',
  icon: '',
  displayOrder: 0,
  isActive: true,
  parentId: undefined,
  sysmodule: undefined,
  isGroup: false,
  children: [],
}

export type SaveMenuDto = {
  id: number
  viName: string,
  enName: string,
  path: string
  icon?: string
  displayOrder: number
  isActive: boolean
  parentId?: number
  sysmodule: string
  isGroup: boolean
}

export const defaultSaveMenuDto: SaveMenuDto = {
  id: 0,
  viName: '',
  enName: '',
  path: '',
  icon: '',
  displayOrder: 0,
  isActive: true,
  parentId: undefined,
  sysmodule: '',
  isGroup: false,
}

File: Core.Domain/Entities/System/Menu.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

// ...existing using statements...

namespace Core.Domain.Entities.System;

[Table("menus")]
public class Menu : BaseEntity<int>
{
    [Required][StringLength(100)] public string ViName { get; set; } = string.Empty;
    [Required][StringLength(100)] public string EnName { get; set; } = string.Empty;
    [Required][StringLength(200)] public string Path { get; set; } = string.Empty;

    [Required][StringLength(100)] public string? Icon { get; set; }

    public int DisplayOrder { get; set; }

    public bool IsActive { get; set; } = true;

    public int? ParentId { get; set; }

    [StringLength(100)] public string? Sysmodule { get; set; }
    public bool IsGroup { get; set; }
}